### ISSUE TITLE
Unable to read non_hadoop_lz4_compressed.parquet

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#3586]
https://github.com/apache/parquet-testing/tree/master/data has a bunch of parquet files for testing purposes.

DH is able to read [hadoop_lz4_compressed.parquet](https://github.com/apache/parquet-testing/blob/master/data/hadoop_lz4_compressed.parquet), but not [non_hadoop_lz4_compressed.parquet](https://github.com/apache/parquet-testing/blob/master/data/non_hadoop_lz4_compressed.parquet).

```
java.io.EOFException: null
        at java.io.DataInputStream.readFully(DataInputStream.java:202)
        at java.io.DataInputStream.readFully(DataInputStream.java:170)
        at org.apache.parquet.bytes.BytesInput$StreamBytesInput.toByteArray(BytesInput.java:286)
        at org.apache.parquet.bytes.BytesInput.copy(BytesInput.java:202)
        at io.deephaven.parquet.compress.DeephavenCompressorAdapterFactory$CodecWrappingCompressorAdapter.decompress(DeephavenCompressorAdapterFactory.java:137)
        at io.deephaven.parquet.base.ColumnPageReaderImpl.readDataPage(ColumnPageReaderImpl.java:224)
        at io.deephaven.parquet.base.ColumnPageReaderImpl.materialize(ColumnPageReaderImpl.java:90)
        at io.deephaven.parquet.table.pagestore.topage.ToPage.getResult(ToPage.java:57)
        at io.deephaven.parquet.table.pagestore.topage.ToPage.toPage(ToPage.java:84)
        at io.deephaven.parquet.table.pagestore.ColumnChunkPageStore.toPage(ColumnChunkPageStore.java:91)
        at io.deephaven.parquet.table.pagestore.VariablePageSizeColumnChunkPageStore.getPage(VariablePageSizeColumnChunkPageStore.java:122)
        at io.deephaven.parquet.table.pagestore.VariablePageSizeColumnChunkPageStore.getPageContaining(VariablePageSizeColumnChunkPageStore.java:161)
        at io.deephaven.parquet.table.pagestore.VariablePageSizeColumnChunkPageStore.getPageContaining(VariablePageSizeColumnChunkPageStore.java:20)
        at io.deephaven.engine.page.PageStore.fillChunk(PageStore.java:75)
        at io.deephaven.parquet.table.region.ParquetColumnRegionBase.fillChunk(ParquetColumnRegionBase.java:50)
        at io.deephaven.engine.page.PageStore.fillChunk(PageStore.java:79)
        at io.deephaven.engine.table.impl.sources.regioned.RegionedColumnSourceBase.fillChunk(RegionedColumnSourceBase.java:51)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.getSnapshotDataAsChunkList(ConstructSnapshot.java:1477)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.serializeAllTable(ConstructSnapshot.java:1358)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.lambda$constructBackplaneSnapshotInPositionSpace$2(ConstructSnapshot.java:596)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1089)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1021)
        at io.deephaven.engine.table.impl.remote.ConstructSnapshot.constructBackplaneSnapshotInPositionSpace(ConstructSnapshot.java:601)
        at io.deephaven.server.barrage.BarrageMessageProducer.getSnapshot(BarrageMessageProducer.java:2143)
        at io.deephaven.server.barrage.BarrageMessageProducer.updateSubscriptionsSnapshotAndPropagate(BarrageMessageProducer.java:1269)
        at io.deephaven.server.barrage.BarrageMessageProducer$UpdatePropagationJob.run(BarrageMessageProducer.java:945)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:162)
        at org.jpy.PyLib.callAndReturnObject(PyLib.java:-2)
        at org.jpy.PyObject.call(PyObject.java:449)
        at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
        at java.lang.Thread.run(Thread.java:829)
```

This is _not_ related to https://github.com/deephaven/deephaven-core/issues/3585.

Note, I've found DuckDB from this comment https://stackoverflow.com/a/74260936 a useful tool in examining parquet metadata: 
```
D SELECT * FROM PARQUET_SCHEMA('hadoop_lz4_compressed.parquet');
┌─────────────────────────────────────────────────────┬─────────┬────────────┬─────────────┬─────────────────┬──────────────┬────────────────┬───────┬───────────┬──────────┬──────────────┐
│                      file_name                      │  name   │    type    │ type_length │ repetition_type │ num_children │ converted_type │ scale │ precision │ field_id │ logical_type │
│                       varchar                       │ varchar │  varchar   │   varchar   │     varchar     │    int64     │    varchar     │ int64 │   int64   │  int64   │   varchar    │
├─────────────────────────────────────────────────────┼─────────┼────────────┼─────────────┼─────────────────┼──────────────┼────────────────┼───────┼───────────┼──────────┼──────────────┤
│ hadoop_lz4_compressed.parquet │         │ BOOLEAN    │ 0           │ REQUIRED        │            3 │ UTF8           │     0 │         0 │        0 │              │
│ hadoop_lz4_compressed.parquet │ c0      │ INT64      │ 0           │ REQUIRED        │            0 │ UTF8           │     0 │         0 │        0 │              │
│ hadoop_lz4_compressed.parquet │ c1      │ BYTE_ARRAY │ 0           │ REQUIRED        │            0 │ UTF8           │     0 │         0 │        0 │              │
│ hadoop_lz4_compressed.parquet │ v11     │ DOUBLE     │ 0           │ OPTIONAL        │            0 │ UTF8           │     0 │         0 │        0 │              │
└─────────────────────────────────────────────────────┴─────────┴────────────┴─────────────┴─────────────────┴──────────────┴────────────────┴───────┴───────────┴──────────┴──────────────┘

D SELECT row_group_id, file_offset, path_in_schema, type, compression, encodings, index_page_offset, dictionary_page_offset, data_page_offset FROM PARQUET_METADATA('hadoop_lz4_compressed.parquet');
┌──────────────┬─────────────┬────────────────┬────────────┬─────────────┬───────────────────────────────────┬───────────────────┬────────────────────────┬──────────────────┐
│ row_group_id │ file_offset │ path_in_schema │    type    │ compression │             encodings             │ index_page_offset │ dictionary_page_offset │ data_page_offset │
│    int64     │    int64    │    varchar     │  varchar   │   varchar   │              varchar              │       int64       │         int64          │      int64       │
├──────────────┼─────────────┼────────────────┼────────────┼─────────────┼───────────────────────────────────┼───────────────────┼────────────────────────┼──────────────────┤
│            0 │           4 │ c0             │ INT64      │ LZ4         │ BIT_PACKED, PLAIN_DICTIONARY      │                 0 │                      0 │                4 │
│            0 │         116 │ c1             │ BYTE_ARRAY │ LZ4         │ BIT_PACKED, PLAIN_DICTIONARY      │                 0 │                      0 │              116 │
│            0 │         195 │ v11            │ DOUBLE     │ LZ4         │ BIT_PACKED, PLAIN_DICTIONARY, RLE │                 0 │                      0 │              195 │
└──────────────┴─────────────┴────────────────┴────────────┴─────────────┴───────────────────────────────────┴───────────────────┴────────────────────────┴──────────────────┘

D SELECT * FROM PARQUET_SCHEMA('non_hadoop_lz4_compressed.parquet');
┌─────────────────────────────────────────────────────────┬─────────┬────────────┬─────────────┬─────────────────┬──────────────┬────────────────┬───────┬───────────┬──────────┬──────────────┐
│                        file_name                        │  name   │    type    │ type_length │ repetition_type │ num_children │ converted_type │ scale │ precision │ field_id │ logical_type │
│                         varchar                         │ varchar │  varchar   │   varchar   │     varchar     │    int64     │    varchar     │ int64 │   int64   │  int64   │   varchar    │
├─────────────────────────────────────────────────────────┼─────────┼────────────┼─────────────┼─────────────────┼──────────────┼────────────────┼───────┼───────────┼──────────┼──────────────┤
│ non_hadoop_lz4_compressed.parquet │ schema  │ BOOLEAN    │ 0           │ REQUIRED        │            3 │ UTF8           │     0 │         0 │        0 │              │
│ non_hadoop_lz4_compressed.parquet │ c0      │ INT64      │ 0           │ OPTIONAL        │            0 │ UTF8           │     0 │         0 │        0 │              │
│ non_hadoop_lz4_compressed.parquet │ c1      │ BYTE_ARRAY │ 0           │ OPTIONAL        │            0 │ UTF8           │     0 │         0 │        0 │              │
 | non_hadoop_lz4_compressed.parquet │ v11     │ DOUBLE     │ 0           │ OPTIONAL        │            0 │ UTF8           │     0 │         0 │        0 │              │
└─────────────────────────────────────────────────────────┴─────────┴────────────┴─────────────┴─────────────────┴──────────────┴────────────────┴───────┴───────────┴──────────┴──────────────┘


D SELECT row_group_id, file_offset, path_in_schema, type, compression, encodings, index_page_offset, dictionary_page_offset, data_page_offset FROM PARQUET_METADATA('non_hadoop_lz4_compressed.parquet');
┌──────────────┬─────────────┬────────────────┬────────────┬─────────────┬──────────────────────────────┬───────────────────┬────────────────────────┬──────────────────┐
│ row_group_id │ file_offset │ path_in_schema │    type    │ compression │          encodings           │ index_page_offset │ dictionary_page_offset │ data_page_offset │
│    int64     │    int64    │    varchar     │  varchar   │   varchar   │           varchar            │       int64       │         int64          │      int64       │
├──────────────┼─────────────┼────────────────┼────────────┼─────────────┼──────────────────────────────┼───────────────────┼────────────────────────┼──────────────────┤
│            0 │         107 │ c0             │ INT64      │ LZ4         │ PLAIN_DICTIONARY, PLAIN, RLE │                 0 │                      4 │               36 │
│            0 │         269 │ c1             │ BYTE_ARRAY │ LZ4         │ PLAIN_DICTIONARY, PLAIN, RLE │                 0 │                    199 │              228 │
│            0 │         441 │ v11            │ DOUBLE     │ LZ4         │ PLAIN_DICTIONARY, PLAIN, RLE │                 0 │                    333 │              369 │
└──────────────┴─────────────┴────────────────┴────────────┴─────────────┴──────────────────────────────┴───────────────────┴────────────────────────┴──────────────────┘
```

**Comment by devinrsmith** [comment]
Based on the LZ4_RAW comment in https://github.com/apache/parquet-testing/commit/d914f9d289488c7db1759d7a88a4a1b8f062c7dd, this *may* actually be #3585

**Comment by devinrsmith** [comment]
I'm beginning to think this file is technically invalid, but the other implementations have workarounds to hack it. I _think_ this is really an LZ4_RAW compressed file, but is marked LZ4.

**Comment by malhotrashivam** [comment]
As suspected by Devin, [non_hadoop_lz4_compressed.parquet](https://github.com/apache/parquet-testing/blob/master/data/non_hadoop_lz4_compressed.parquet) is actually LZ4_RAW compressed and not LZ4 but the metadata says the file is compressed with LZ4. Our parquet reading code picks up compression type from metadata and therefore fails while reading. On the other hand, [hadoop_lz4_compressed.parquet](https://github.com/apache/parquet-testing/blob/master/data/hadoop_lz4_compressed.parquet) is actually LZ4 compressed and tagged as LZ4 only. Therefore, works with our reading code.

Pyarrow can still read both files. I tested reading the metadata for these files with pyarrow and it reads the compression type as 'UNKNOWN'. It can mean that pyarrow is inferring the compression type from the data instead of reading it from metadata for these files, and thus correctly decompressing them.

I don't think we need to add additional logic to handle such cases right now.

**Comment by malhotrashivam** [comment]
Resolved as part of #4480
